### PR TITLE
bug(NestedInput): handleClick and prevent default possible fix

### PR DESCRIPTION
**Pull request by katiejonesyo**

### CODE REVIEW

File: packages/core/src/types/index.ts [hunk 1]
```diff
@@ -152,6 +152,7 @@ export interface ButtonProps {
   style?: React.CSSProperties;
   variant?: ButtonVariant;
   onClick?: (e?: React.MouseEvent) => unknown;
+  preventDefault?: boolean;
```
Comment (erwstout) [inline packages/core/src/types/index.ts:155]: Where is this coming into play? It doesn't appear to be used

File: packages/core/src/types/index.ts [hunk 2]
```diff
@@ -354,6 +355,8 @@ export interface DropdownStyleProps {
 export interface NestedInputProps extends BaseFormikFieldsWithHandleChange {
   buttonText: string;
   placeholder: string;
+  preventDefault: boolean;
```
Comment (erwstout) [inline packages/core/src/types/index.ts:358]: Where is this coming into play? It doesn't appear to be used

File: packages/core/src/types/index.ts [hunk 3]
```diff
@@ -354,6 +355,8 @@ export interface DropdownStyleProps {
...
+  handleClick: React.MouseEventHandler<HTMLButtonElement>;
```
Comment (erwstout) [inline packages/core/src/types/index.ts:359]: Believe this should be optional, since in other cases we may not want to pass a handleClick function

**Review by erwstout — CHANGES_REQUESTED** [review]
Tests are failing as well since we updated the HTML structure. From the root of the repo running `yarn test -u` will update the snapshots that are currently failing. 

Also - for everything in the UI package we should be updating the documentation (storybook folder) to add the new `handleClick` prop on the Nested Input as well as adding a test that the click function works. Take a look at `Button.test.tsx` for examples of how to do click testing.

**Review by JimBurch — APPROVED** [review]

**Pull request closed without merging**